frame/collective/src/lib.rs

```diff
@@ -168,12 +175,149 @@ decl_error! {
+	) -> Weight {
+		db.reads(1) // read members for `is_member`
+			.saturating_add(23_000_000) // constant
+			.saturating_add(3_000 * MAX_ASSUMED_PROPOSAL_BYTES) // B
+			.saturating_add(120_000 * members.into()) // M
+			.saturating_add(proposal.into()) // P
+	}
+
+	/// Calculate the weight for `propose` if the proposal is executed straight away (`threshold < 2`).
+	pub(crate) fn propose_execute(
+		db: RuntimeDbWeight,
+		members: impl Into<Weight>,
+		proposal: impl Into<Weight>,
+	) -> Weight {
+		db.reads(2) // `is_member` + `contains_key`
+			.saturating_add(29_000_000) // constant
+			.saturating_add(3_000 * MAX_ASSUMED_PROPOSAL_BYTES) // B
+			.saturating_add(220_000 * members.into()) // M
+			.saturating_add(proposal.into()) // P1
+	}
+
+	/// Calculate the weight for `propose` if the proposal is put up for a vote (`threshold >= 2`).
+	pub(crate) fn propose_proposed(
+		db: RuntimeDbWeight,
+		members: impl Into<Weight>,
+		proposals: impl Into<Weight>,
+	) -> Weight {
+		db.reads(2) // `is_member` + `contains_key`
+			+ db.reads_writes(2, 4) // `proposal` insertion
+			+ 50_000_000 // constant
+			+ 6_000 * MAX_ASSUMED_PROPOSAL_BYTES // B
+			+ 110_000 * members.into() // M
+			+ 510_000 * proposals.into() // P2
+	}
+
+	/// Calculate the weight for `vote`.
+	pub(crate) fn vote(
+		db: RuntimeDbWeight,
+		members: impl Into<Weight>,
+	) -> Weight {
+		db.reads(1) // read `Members`
+			+ db.reads_writes(1, 1) // mutate `Voting`
+			+ 30_000_000 // constant
+			+ 500_000 * members.into() // M
 	}
+
+	/// Calculate the weight for `close`.
+	pub(crate) fn close(
+		db: RuntimeDbWeight,
+		members: impl Into<Weight> + Copy,
+		proposal_weight: impl Into<Weight>,
+		proposals: impl Into<Weight>,
+	) -> Weight {
+		close_without_finalize(db, members)
+			.saturating_add(db.reads(1)) // `Prime`
+			.saturating_add(db.writes(1)) // `Proposals`
+			.saturating_add(db.writes(1)) // `Voting`
+			.saturating_add(proposal_weight.into()) // P1
+			.saturating_add(490_000 * proposals.into()) // P2
+	}
+
+	/// Calculate the weight for `close` without the call to `finalize_proposal`.
+	pub(crate) fn close_without_finalize(
+		db: RuntimeDbWeight,
+		members: impl Into<Weight>,
+	) -> Weight {
+		db.reads(3) // `Members`, `Voting`, `ProposalOf`
+			+ 66_000_000 // constant
+			+ 8_000 * MAX_ASSUMED_PROPOSAL_BYTES // B
+			+ 250_000 * members.into() // M
+	}
+}
+
+/// Adjust the weight of a dispatch call result by mapping the weight with a closure.
+///
+/// Will return `Ok` without regard to the underlying result.
+///
+/// Usage:
+/// ```ignore
+/// let res = fold_result_weight(result, |weight| weight + 40_000);
+/// ```
+fn fold_result_weight<F>(
+    result: DispatchResultWithPostInfo,
+    closure: F,
+) -> DispatchResultWithPostInfo
+where
+    F: FnOnce(Weight) -> Weight,
+{
+    match result {
+        Ok(post_info) => Ok(post_info.actual_weight.map(closure).into()),
+        Err(err) => Ok(err.post_info.actual_weight.map(closure).into()),
+    }
```

Comment:
why not use get_result_weight here. (also I think considering get_result_weight this function isn't that useful but I'm OK anyway).